src/view/tree_structure_tracking.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{
+    id::Id,
+    view::ElementsSplice,
+    widget::{ChangeFlags, Pod},
+};
+use xilem_core::VecSplice;
+
+use super::Cx;
+
+#[derive(Clone, Copy, Debug)]
+pub enum SpliceMutation {
+    Add(Id),
+    Change(Id),
+    Delete(usize),
+    Skip(usize),
+}
+
+/// An ElementsSplice that monitors the tree structure by maintaining a mutation log
+pub struct TreeStructureTrackingSplice<'a, 'b, 'c> {
+    current_child_id: Option<Id>,
+    splice: VecSplice<'a, 'b, Pod>,
+    mutations: &'c mut Vec<SpliceMutation>,
+}
```

Comment:
The name feels a bit long. Maybe rename to `TreeStructureSplice`? (Not very important.)